File: Data Structures & Algorithms/merge-two-sorted-linked-lists/submission-0.js
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     constructor(val = 0, next = null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */

class Solution {
    /**
     * @param {ListNode} list1
     * @param {ListNode} list2
     * @return {ListNode}
     */
    mergeTwoLists(list1, list2) {
        let p1 = list1;
        let p2 = list2;
        let head = new ListNode(0, null)
        let firstHead = head;
        while (p1 && p2) {
            if (p1.val < p2.val) {
                head.next = p1;
                p1 = p1.next;
            } else {
                head.next = p2;
                p2 = p2.next;
            }
            head = head.next;
        }

        if (p1) {
            head.next = p1;
        } else {
            head.next = p2;
        }


        return firstHead.next;
    }
}

File: Data Structures & Algorithms/maximum-twin-sum-of-a-linked-list/submission-0.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public int pairSum(ListNode head) {
        int L = 0, R = 0;
        List<Integer> arr = new ArrayList<>();
        while(head!=null) {
            arr.add(head.val);
            head = head.next;
        }
        R = arr.size()-1;
        int max = Integer.MIN_VALUE;
        while (L < R) {
            int sum = arr.get(L++) + arr.get(R--);
            max = Math.max(max, sum);
        }   
        return max;
    }
}